src/components/Widget/Widget.js

```diff
@@ -61,13 +63,42 @@ class Widget extends React.Component {
         })
     }
 
+    // https://mhnpd.github.io/react-loader-spinner/?selectedKind=Loader&selectedStory=Oval&full=0&addons=0&stories=1&panelRight=0
+    // Link for loader types
+    loader = () => {
+        return (
+            <div className={"loader-widget"}>
+                <Loader 
+                    type="Oval"
+                    color="red"
+                    height="100"	
+                    width="100"
+                />
+            </div>
+        )
+    }
+
+
     getCurrentSettings = () => this.state.currentSettings
 
     defaultSettings = () => {
         this.setState({currentSettings: this.state.defaultSettings})
     }
 
     makeComponent = (visibility, id) => {
+        // If state.loading = true, then we display the loader else the widget
+        if (this.state.loading) {
+            return (
+                <div>
+                    {/* We need to clone the element, but we don't display it, Not sure what this does, probably a better solution than this */}
+                    <div className={"dashboard-widget-content none"} id={id} style={{display: 'none'}}>
```

Comment:
We already have something for this. You can simply add `hide` to the `className`, after which it should be hidden.